### ISSUE TITLE
🐛 Permission overrides do not work properly if you have one role override

### ISSUE THREAD

**Issue by MarcusOtter** [MarcusOtter/discord-needle#295]
### Describe the bug

Basically, in this case Needle says "You do not have permission to perform this action" to thread authors (people that started the thread).

![image](https://user-images.githubusercontent.com/35617441/188331072-d464aef4-eabb-473d-9e4a-6a0237b7df7d.png)

For some reason Needle assumes that it should not use default permissions in this case, or something. It doesn't realize that the user is unaffected by the overrides.

### Steps to reproduce the bug

1. Add one role override to all Needle commands set to "Allow" (green check mark). See image above.
2. Edit a title of a thread as a thread author (without admin perms). You should also not have any roles.
3. See "You do not have permission to perform this action"

### Expected behavior

An unrelated override should not affect the permission calculation for the user. They should be allowed to rename the thread.

**Comment by MarcusOtter** [comment]
Massive thanks to @Aelweak for reporting this bug with great detail 🏆

**Comment by Madis0** [comment]
Can confirm, an user reported not being able to use /title but able to use the button.

**Comment by MarcusOtter** [comment]
That actually sounds like it could be a different problem @Madis0 as the `/title` command check does not do any special permission checks on Needle's side, but the button does (and the problem in this case is with the button, I think). Users that can send the command should be able to press the button. Users that can not send the command should not be able to press the button. Did you have the same setup as this issue with the permission overrides, and did it go away after you removed the role override? If not, could you open a separate issue?

**Comment by MarcusOtter** [comment]
@cdtinney Yes, Needle allows for this behaviour as long as you do not change the permission overrides, as that is the default behavior. If you add permission overrides it will stop working. Any reason you need to override permissions?
